### ISSUE TITLE
Secure Desktop option incompatible with Windows 11 22H2

### ISSUE THREAD

**Issue by KnPk11** [veracrypt/VeraCrypt#1014]
# Expected behavior
When using the Secure Desktop for password entry, VeraCrypt switches out of Windows Explorer while typing the password, and upon completion it returns back to Windows Desktop.

# Observed behavior
VeraCrypt's Secure Desktop is now almost unusable, as it hangs on a white screen after trying to mount a container. 
This may happen (if unlucky) on the very first mounting attempt, or more often on the 2nd or 3rd attempt. 
So you may be able to mount 1-2 containers initially, but later get into the trap of being stuck on the Secure Desktop's white screen, unable to interact with anything but the task manager, and having to log off and log back on (losing any unsaved work). 

# Steps to reproduce
The easiest way to reproduce this on a Windows 11 22H2 system is to:
1) Select any container for mounting
2)  Keep entering the wrong password and you should soon hit the issue soon

Or just try mounting 3 different containers sequentially and you'll likely hit the issue.

# Your Environment
This happenes to me on three different machines: my gaming PC, my work laptop and my mini-PC, all running latest version of Windows 11.

**VeraCrypt version:** 1.25.9

**Operating system and version:** Windows 11 Home/Pro 22H2 (build 22621.963)

**System type:** 64-bit

# Misc.
As discussed in [this superuser article](https://superuser.com/questions/1745373/windows-11-screen-goes-black-after-using-secure-desktop-for-entering-a-passwor)

**Comment by Radium-bit** [comment]
Yes, there are similar issues raised on [sourceforge](https://sourceforge.net/p/veracrypt/discussion/general/thread/14bf57276d/?limit=25#4c92) .

**Comment by QS8** [comment]
VeraCrypt version: 1.25.9
Operating system: Windows 11 Pro 22H2 (build 22621.1848)
System type: 64-bit

A somewhat different behavior, which might be related to the issue described above ...
On one occasion, (which was NOT the first time that I had mounted the encrypted container since boot up,) I chose a particular encrypted container from the Favorites in Veracrypt, which brought up the Secure Desktop, and then successfully entered the password for the encrypted container. The container mounted successfully, and the Veracrypt window was displayed on the monitor for a few hundred milliseconds probably. But then immediately jumped back to the Secure Desktop. I clicked Cancel, but then the Secure Desktop appeared again. and again and again. I realized that Veracrypt was looping thru the encrypted containers which are listed in the Favorites which I had set up in Veracrypt on that machine. So, after I clicked Cancel for each of the encrypted containers which was listed in Favorites, then the Veracrypt window was displayed on the monitor. (I think it was each and every one of them, but am not able to confirm that.) And everything was okay, with the first encrypted container still mounted successfully, and the Veracrypt window displaying it as such.

**Comment by AlainCo** [comment]
Since few months, I have a similar problem. 
I mount the preferred volumes with secure desktop, and it is switching between the secure desktop and the passphrase typing, and the main desktop and the progressbar of Veracrypt.
No crash, always happening, just annoying.
It is strange that the problem appeared recently, not with 22H2. same today after updating to 23H2.
Maybe it is linked to a new video driver for a USB-VGA device(MindShow USB Display https://github.com/MindShow/USBDisplay/tree/main ) , but I have uninstalled the devices and the driver, and nothing is better.

**Comment by idrassi** [comment]
A fix has been implemented and version 1.26.10  has been published to [Sourceforge Nightly Builds](https://sourceforge.net/projects/veracrypt/files/VeraCrypt%20Nightly%20Builds/Windows/).
A user on Sourceforge confirmed that it is working properly now: https://sourceforge.net/p/veracrypt/discussion/general/thread/67d26e9caf/
The release notes mention the fix: https://veracrypt.fr/en/Release%20Notes.html

Please give it a try and report if you still encounter any issues.

**Comment by AlainCo** [comment]
Thanks, I missed this release.
 I've installed version 1.26.10, with the MindShow USBDisplay MSDisplay_Windows_V2.0.1.7.3.exe driver/tray (no USB device plugged), 
and there is still the problem. Screen is switching between secure and normal desktop....

 It may be a different problem, quite specific to the driver... the UAC works well, but not the Veracrypt password.

**Comment by spiralofhope** [comment]
With the new version, I also still saw the secure desktop problem after my second unmount/mount.

I'm not sure how to test in a safe way, and I'm not sure how I ought to reproduce it and what information to provide.

**Comment by leonard84** [comment]
I'm still experiencing this problem with version 1.26.7

**Comment by QS8** [comment]
In v 1.26.15, after using it about a month or so, it seems that this bug is solved. Works fine now with the 'Use Secure Desktop for password entry' option turned on in Veracrypt Preferences.
